test/threads.jl

```diff
@@ -141,3 +141,41 @@ end
 @test_throws TypeError Atomic{Bool}
 @test_throws TypeError Atomic{BigInt}
 @test_throws TypeError Atomic{Float64}
+
+# Test atomic fences
+type CommBuf
+    barrier::Atomic{Int}
+    data::Vector{Float64}
+    correct::Bool
+    CommBuf() = new(Atomic{Int}(0), Vector{Float64}(100), false)
+end
+function test_fence_write(commbuf::CommBuf)
+    sleep(0.1)
+    for i in eachindex(commbuf.data)
+        commbuf.data[i] = i
+    end
+    atomic_fence()
+    commbuf.barrier[] = 1
```

Comment:
I can't think of a way to write a fence test.

Fences are only interesting for stores or loads that are not already ordered. Our atomic operators are currently all ordered. On the other hand, non-atomic loads and stores cannot be directly observed from another thread (as this is undefined), and ensuring the proper ordering requires ordered atomic load/stores, which then remove the need for a fence.

I think acquire-release fences only make sense if there are unordered atomic operations. We currently don't provide these in Julia.

I am thus changing the fence to a sequential-consistency fence.